### PR TITLE
Update user_session table

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**

### CODE REVIEW

File: lib/dbservice/sessions/user_session.ex [hunk 1]
```diff
@@ -5,17 +5,18 @@ defmodule Dbservice.Sessions.UserSession do
...
     field :timestamp, :utc_datetime
     field :data, :map
-    field :type, :string
+    field :user_activity_type, :string
+    field :user_activity_sub_type, :string
 
     timestamps()
 
-    belongs_to :session_occurrence, SessionOccurence
+    belongs_to :session, Session
```
Comment (pritamps) [inline lib/dbservice/sessions/user_session.ex:19]: What is the logic here? User interacts with session occurrence, not session no?

File: priv/repo/migrations/20240525065205_update_user_session.exs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+defmodule Dbservice.Repo.Migrations.UpdateUserSession do
+  use Ecto.Migration
+
+  def change do
+    alter table(:user_session) do
+      add :user_activity_type, :string
```
Comment (pritamps) [inline priv/repo/migrations/20240525065205_update_user_session.exs:6]: `subtype` also needs to be added here?

**Review by pritamps — CHANGES_REQUESTED** [review]

File: priv/repo/migrations/20240525065205_update_user_session.exs (hunk 2 above)
Comment (sudheshna-donthineni) [reply priv/repo/migrations/20240525065205_update_user_session.exs:6]: User does not have a `subtype`. A session does.

File: lib/dbservice/sessions/user_session.ex [hunk 3]
```diff
@@ -5,31 +5,42 @@ defmodule Dbservice.Sessions.UserSession do
...
+    belongs_to :session_occurrence, SessionOccurrence
     belongs_to :user, User
   end
 
   @doc false
   def changeset(user_session, attrs) do
+    IO.inspect(attrs)
+
```
Comment (Bahugunajii) [inline lib/dbservice/sessions/user_session.ex:27]: Let's remove debugging statements

File: priv/repo/migrations/20240525065205_update_user_session.exs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      remove :session_occurrence_id
+      add :session_occurrence_id, references(:session_occurrence)
```
Comment (Bahugunajii) [inline priv/repo/migrations/20240525065205_update_user_session.exs:8]: What's the point of removing session occurrence and adding it again?

File: priv/repo/migrations/20240525065205_update_user_session.exs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+      add :session_id, references(:session)
+      remove :user_id
+      add :user_id, references(:user)
```
Comment (Bahugunajii) [inline priv/repo/migrations/20240525065205_update_user_session.exs:11]: Same question here

**Review by Bahugunajii — CHANGES_REQUESTED** [review]

**Comment by Bahugunajii** [comment]
Let's also fix the failing CI checks

File: priv/repo/migrations/20240525065205_update_user_session.exs (hunk 4 above)
Comment (sudheshna-donthineni) [reply priv/repo/migrations/20240525065205_update_user_session.exs:8]: It didnt have any reference, was storing it as a string. I could probably do update also.

Comment (Bahugunajii) [reply priv/repo/migrations/20240525065205_update_user_session.exs:8]: `session_occurrence_id` did had reference. Only user_id didn't had

File: priv/repo/migrations/20240525065205_update_user_session.exs (hunk 5 above)
Comment (sudheshna-donthineni) [reply priv/repo/migrations/20240525065205_update_user_session.exs:9]: Like you mentioned, `user_id` did not have any reference

File: priv/repo/migrations/20240525065205_update_user_session.exs (hunk 4 above)
Comment (sudheshna-donthineni) [reply priv/repo/migrations/20240525065205_update_user_session.exs]: You're right, my bad, removed it.

**Review by Bahugunajii — APPROVED** [review]

**Pull request merged**